apps/core/lib/core/snitch/data/model/payment/card_payment.ex

```diff
@@ -0,0 +1,95 @@
+defmodule Core.Snitch.Data.Model.CardPayment do
+  @moduledoc """
+  CardPayment API and utilities.
+
+  `CardPayment` is a concrete payment subtype in Snitch. By `create/2`ing a
+  CardPayment, the supertype Payment is automatically created in the same
+  transaction.
+
+  > For other supported payment sources, see
+    `Core.Snitch.Data.Schema.PaymentMethod`
+  """
+  use Core.Snitch.Data.Model
+
+  @doc """
+  Creates both `Payment` and `CardPayment` records in a transaction for Order
+  represented by `order_id`.
+
+  * `payment_params` are validated using
+    `Core.Snitch.Data.Schema.Payment.changeset/3` with the `:create` action and
+    because `slug` and `order_id` are passed explicitly to this function,
+    they'll be ignored if present in `payment_params`.
+  * `card_params` are validated using
+  `Core.Snitch.Data.Schema.CardPayment.changeset/3` with the `:create` action.
+  """
+  @spec create(String.t(), non_neg_integer(), map, map) ::
+          {:ok, %{card_payment: Schema.CardPayment.t(), payment: Schema.Payment.t()}}
+          | {:error, Ecto.Changeset.t()}
+  def create(slug, order_id, payment_params, card_params) do
+    payment = struct(Schema.Payment, payment_params)
+    card_method = Model.PaymentMethod.get_card()
+
+    more_payment_params = %{
+      order_id: order_id,
+      payment_type: "ccd",
```

Comment:
Can we a constant file for this from which we can pick using the key ?
its kinda getting repeated everywhere, would be difficult to make change everywhere in future if there is a need.  😄

Reply:
Sure, that makes a lot of sense.
1. `seed/payment_method.exs` Will read that file and insert accordingly.
2. Code in other places (like here) can do the same.

IMO this is file would be like a config file, so let's add it later (I opened #20 on this)